src/generate.ts

```diff
@@ -0,0 +1,192 @@
+import type {
+	AbstractSqlField,
+	AbstractSqlModel,
+	AbstractSqlTable,
+	InNode,
+	Relationship,
+	RelationshipInternalNode,
+	RelationshipLeafNode,
+} from '@balena/abstract-sql-compiler';
+import {
+	odataNameToSqlName,
+	sqlNameToODataName,
+} from '@balena/odata-to-abstract-sql';
+import { replaceResultTransformer, TemplateTag } from 'common-tags';
+
+type RequiredModelSubset = Pick<
+	AbstractSqlModel,
+	'tables' | 'relationships' | 'synonyms'
+>;
+
+const typeHelpers = `import type { Types } from '@balena/abstract-sql-to-typescript';\n`;
+
+const trimNL = new TemplateTag(
+	replaceResultTransformer(/^[\r\n]*|[\r\n]*$/g, ''),
+);
+
+const modelNameToCamelCaseName = (s: string): string =>
+	s
+		.split(/[ -]/)
+		.map((p) => p[0].toLocaleUpperCase() + p.slice(1))
+		.join('');
+
+const getReferencedInterface = (modelName: string, mode: Mode) =>
+	`${modelNameToCamelCaseName(modelName)}['${mode}']`;
+
+const sqlTypeToTypescriptType = (
+	m: RequiredModelSubset,
+	f: AbstractSqlField,
+	mode: Mode,
+): string => {
+	if (!['ForeignKey', 'ConceptType'].includes(f.dataType) && f.checks) {
+		const inChecks = f.checks.find(
+			(checkTuple): checkTuple is InNode => checkTuple[0] === 'In',
+		);
+		if (inChecks) {
+			const [, , ...allowedValues] = inChecks;
```

Comment:
How about using `slice(2)` here instead of destructuring and creating shallow copies?